### PR TITLE
chore(catalog): clean up CatalogTable, use only Entity

### PR DESCRIPTION
**Pull request by freben**
#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

File: plugins/catalog/src/components/CatalogTable/CatalogTable.tsx [hunk 1]
```diff
@@ -13,33 +13,33 @@
...
 import { entityRoute } from '../../routes';
 
 const columns: TableColumn[] = [
   {
     title: 'Name',
-    field: 'name',
+    field: 'metadata.name',
     highlight: true,
-    render: (componentData: any) => (
+    render: (entity: any) => (
```
Comment (shmidt-i) [inline plugins/catalog/src/components/CatalogTable/CatalogTable.tsx:29]: Can be now `entity:Entity`, right?

File: plugins/catalog/src/components/CatalogPage/CatalogPage.tsx [hunk 2]
```diff
@@ -89,7 +88,7 @@ export const CatalogPage: FC<{}> = () => {
         hidden: location ? location?.type !== 'github' : true,
```
Comment (hooloovooo) [inline plugins/catalog/src/components/CatalogPage/CatalogPage.tsx:88]: Use `location?.type !== 'github'` here as well? It's a lot cleaner.

**Review by shmidt-i — APPROVED** [review]

**Review by hooloovooo — APPROVED** [review]

File: plugins/catalog/src/components/CatalogPage/CatalogPage.tsx (hunk 2 above)
Comment (freben) [reply plugins/catalog/src/components/CatalogPage/CatalogPage.tsx:88]: Done!

File: plugins/catalog/src/components/CatalogTable/CatalogTable.tsx (hunk 1 above)
Comment (freben) [reply plugins/catalog/src/components/CatalogTable/CatalogTable.tsx:29]: unfortunately not according to the signature of that interface :/

**Pull request merged**